tests/test_core.py

```diff
@@ -7602,20 +7617,19 @@ def setUp(self):
     self.emcc_args = emcc_args[:]
     for k, v in settings.items():
       self.set_setting(k, v)
-    Building.COMPILER_TEST_OPTS += [
-        '-Werror', '-Wno-dynamic-class-memaccess', '-Wno-format',
-        '-Wno-format-extra-args', '-Wno-format-security',
-        '-Wno-pointer-bool-conversion', '-Wno-unused-volatile-lvalue',
-        '-Wno-c++11-compat-deprecated-writable-strings',
-        '-Wno-invalid-pp-token', '-Wno-shift-negative-value'
-    ]
 
-    for arg in self.emcc_args:
-      if arg.startswith('-O'):
-        Building.COMPILER_TEST_OPTS.append(arg) # so bitcode is optimized too, this is for cpp to ll
-      # propagate LLVM machine arguments
-      if arg.startswith('-m'):
-        Building.COMPILER_TEST_OPTS.append(arg)
+    # keep a copy of the settings for this test mode, so we can know
+    # them even in tests that modify settings
+    self.test_mode_settings = settings
```

Comment:
Do you need this extra copy?   Can't you just just `self.get_setting()` where you need to test for settings?

Reply:
Hmm, almost - the question asked here is "does this test have a change on that setting?" `get_setting` will always return a value (changed, or the default). How about adding a `has_changed_setting`? I pushed a commit with that now, let me know what you think.